serializers/src/main/java/io/pravega/schemaregistry/serializers/Codecs.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.serializers;
+
+import io.pravega.schemaregistry.codec.Codec;
+import io.pravega.schemaregistry.contract.data.CodecType;
+import lombok.Getter;
+import org.xerial.snappy.Snappy;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Utility class for creating codecs for none, snappy or gzip. 
+ */
+public enum Codecs {
+    None(Constants.NOOP, Constants.NOOP.getCodecType()),
+    GzipCompressor(Constants.GZIP_CODEC, Constants.GZIP_CODEC.getCodecType()), 
+    SnappyCompressor(Constants.SNAPPY_CODEC, Constants.SNAPPY_CODEC.getCodecType());
+
+    @Getter
+    private final Codec codec;
+    @Getter
+    private final CodecType codecType;
+    
+    Codecs(Codec codec, CodecType codecType) {
+        this.codec = codec;  
+        this.codecType = codecType;
+    }
+
+    private static class Noop implements Codec {
+        private static final CodecType CODEC_TYPE_NONE = new CodecType(Constants.NONE);
+
+        @Override
+        public CodecType getCodecType() {
+            return CODEC_TYPE_NONE;
+        }
+
+        @Override
+        public ByteBuffer encode(ByteBuffer data) {
+            return data;
+        }
+
+        @Override
+        public ByteBuffer decode(ByteBuffer data) {
+            return data;
+        }
+    }
+
+    private static class GZipCodec implements Codec {
+        private static final CodecType CODEC_TYPE_GZIP = new CodecType(Constants.APPLICATION_X_GZIP);
+        @Override
+        public CodecType getCodecType() {
+            return CODEC_TYPE_GZIP;
+        }
+        
+        @Override
+        public ByteBuffer encode(ByteBuffer data) throws IOException {
+            ByteArrayOutputStream bos = new ByteArrayOutputStream(data.remaining());
+            GZIPOutputStream gzipOS = new GZIPOutputStream(bos);
+            gzipOS.write(data.array(), data.arrayOffset() + data.position(), data.remaining());
+            gzipOS.close();
+            byte[] compressed = bos.toByteArray();
+            return ByteBuffer.wrap(compressed);
+        }
+
+        @Override
+        public ByteBuffer decode(ByteBuffer data) throws IOException {
+            byte[] array = new byte[data.remaining()];
+            data.get(array);
+
+            ByteArrayInputStream bis = new ByteArrayInputStream(array);
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            GZIPInputStream gzipIS = new GZIPInputStream(bis);
+            byte[] buffer = new byte[1024];
```

Comment:
GZIPInputStream creates an input stream with 512 bytes as buffer. Do we require this buffer?

Reply:
```suggestion
          ByteBufferBackedInputStream bis = new ByteBufferBackedInputStream(data);
return ByteBuffer.wrap(IOUtils.toByteArray(new GZIPInputStream(bis)));
```